18F/calc
# Region 10 Bulk Upload

Main functionality is now complete!

Instead of taking a CSV, this flow can directly take the XLSX spreadsheet that Kelly generates from an export of his Region 10 database. Upon submission of step 2 in this flow, all the existing Region 10 data is purged from the database and the newly uploaded data is saved. To accomplish purging the old data, I added to add a new model `BulkUploadContractSource` for tracking Region 10 data provenance, and associate it with the `Contract` model.

**To Dos**
- [x] import from R10 Excel dump file
- [x] commit migrations once models are finished
- [x] clean up names (`bulk_upload`, `price_list_upload`, `price_list`, etc.)
- [x] use base template, ref #476 

Other stuff to be addressed later:
- Proper cancel behavior that clears session storage (#478)
- Improve UI. I just made it up as I went along but it would be good to have other eyes on it to clean up language/styling/etc.
- In step 1, add a screenshot or something to let the user know which file we're looking for.
- On final step, add ability to download a spreadsheet or CSV of the `bad_rows` so that the admin can see which things _did not_ get added.
- Add JS (and maybe server-side code?) to disable/hide the submit button in step 2 after it is clicked. Since this is such a long process, it would be good to prevent accidental double-submissions.
- Add a spinner in step 2 so the user knows something is happening even though it is taking a while.
- Add a link somewhere (in Admin UI or in CALC header) to the Region 10 bulk upload flow (for now one has to manually visit it at `/data-capture/bulk/region-10/step/1`).
- Examine price list upload and region 10 bulk upload URL schemes.
- Set `BulkUploadContractSource` association in `load_s70` command.
- Return 403 page when non-`is_staff` user attempts to visit bulk upload views. Right now it keeps redirecting to the login flow when a non-staff user logs in with the redirect_url set to a bulk upload step.
- Add file type to `BulkUploadContractSource` so we know if it was a CSV or XLS(X)
